Cabal/Distribution/Simple/Program/Ar.hs

```diff
@@ -83,10 +86,29 @@ createArLibArchive verbosity lbi targetPath files = do
       middle  = initial
       final   = programInvocation ar (finalArgs   ++ extraArgs)
 
-  sequence_
+      oldVersionManualOverride =
+        fromFlagOrDefault False $
+        configArDoesNotSupportAtArguments $
+        configFlags lbi
+      atArgumentNotSupported   =
+        not (arAtFileArgumentSupported (compiler lbi))
+
+      atFileInvocation :: FilePath -> ProgramInvocation
+      atFileInvocation atFile =
+        programInvocation ar $
+        simpleArgs ++ extraArgs ++ ['@' : atFile]
+
+  if oldVersionManualOverride || atArgumentNotSupported
+    then
+      sequence_
         [ runProgramInvocation verbosity inv
         | inv <- multiStageProgramInvocation
                    simple (initial, middle, final) files ]
+    else
+      withTempFile tmpDir "object-files.txt" $ \argsPath argsHandle -> do
```

Comment:
I'm not sure there's a `--keep-tmp-files` flag in cabal-install. There's only `--keep-temp-files` in `cabal haddock`, but that's not relevant I reckon.

Surely I can put the `Ex` version there, but unfortunately I don't see what to pass to it except `defaultTempFileOptions`, which has temp files disabled unconditionally.